### PR TITLE
add new EL string functions

### PR DESCRIPTION
**Pull request by mystiberry-okta**
I also ordered the string functions in alpha order.

@donwoods-okta  and @eklein-okta can you review? Needs to be published by Wed. so RN link will work. THANKS!

### CODE REVIEW

File: _source/_reference/okta_expression_language/index.md [hunk 1]
```diff
@@ -70,21 +70,25 @@ Okta offers a variety of functions to manipulate attributes or properties to gen
...
 | true
- | | | `String.stringContains("This is a test", "doesn'tExist")` | false
-`String.removeSpaces` | (String input) | String | `String.removeSpaces("This is a test")` | Thisisatest
+`String.append` | (String str, String suffix) | String | `String.append("This is", " a test")` | This is a test
 `String.join` | (String separator, String... strings) | String | `String.join(",", "This", "is", "a", "test")` | This,is,a,test
  | | | `String.join("", "This", "is", "a", "test")` | Thisisatest
 `String.len` | (String input) | Integer | `String.len("This")` | 4
-`String.append` | (String str, String suffix) | String | `String.append("This is", " a test")` | This is a test
+`String.removeSpaces` | (String input) | String | `String.removeSpaces("This is a test")` | Thisisatest
+`String.replaceAll`   | (String input, match, replacement) | String | '`String.replaceAll("This is a test", "is", "at")` | "That at a test"
```
Comment (donwoods-okta) [inline _source/_reference/okta_expression_language/index.md]: Oops, my mistake!  The names of the functions got changed in design review, but I forgot to change them in the examples in my wiki.  Fixed now.  They are (and were) correct in the CVD Expressions wiki page.

Anyway, this function should be String.replace, NOT String.replaceAll.  Please fix the name and the example in this line.

File: _source/_reference/okta_expression_language/index.md [hunk 2]
```diff
@@ -70,21 +70,25 @@ Okta offers a variety of functions to manipulate attributes or properties to gen
...
+`String.replaceOnce`   | (String input, match, replacement) | String | '`String.replaceOnce("This is a test", "is", "at")` | "That is a test"
+`String.stringContains` | (String input, String searchString) | Boolean |`String.stringContains("This is a test", "test")`
 | true
+ | | | `String.stringContains("This is a test", "doesn'tExist")` | false
+`String.substring `| (String input, int startIndex, int endIndex) | String | `String.substring("This is a test", 2, 9)` | is is a
+`String.substringAfter` | (String input, String searchString) | String | `String.substringAfter("abc@okta.com", "@")` | okta.com
+`String.substringBefore` | (String input, String searchString) | String | `String.substringBefore("abc@okta.com", "@")` | abc
+`String.stringSwitch` | (String input, String defaultString, String... keyValuePairs) | String | `String.stringSwitch("This is a test", "default", "key1", "value1")`| default
```
Comment (donwoods-okta) [inline _source/_reference/okta_expression_language/index.md]: If the idea was to put these in alphabetical order, then stringSwitch should come before substring.

File: _source/_reference/okta_expression_language/index.md (hunk 1 above)
Comment (mystiberry-okta) [inline _source/_reference/okta_expression_language/index.md]: Done.

**Comment by mystiberry-okta** [comment]
Should this be in the RN for .41? It wasn't listed, but seems like customers might want to know...

**Comment by donwoods-okta** [comment]
:ship:  As for RN, Brian says he tagged this for the Platform RN (did he? or was he thinking of the String.stringSwitch minor change?) and that he defers to your judgment as to whether it should also be in the Product RN.

**Comment by mystiberry-okta** [comment]
he did, no worries! It’s all set!

Mysti
On Oct 11, 2016, at 9:25 AM, donwoods-okta <notifications@github.com<mailto:notifications@github.com>> wrote:


🚢 As for RN, Brian says he tagged this for the Platform RN (did he? or was he thinking of the String.stringSwitch minor change?) and that he defers to your judgment as to whether it should also be in the Product RN.

**Comment by mystiberry-okta** [comment]
waiting to merge until .41 rolls out.

**Pull request merged**